velox/exec/WindowFunction.h

```diff
@@ -15,12 +15,22 @@
  */
 #pragma once
 
+#include "velox/core/Expressions.h"
 #include "velox/exec/WindowPartition.h"
 #include "velox/expression/FunctionSignature.h"
 #include "velox/vector/BaseVector.h"
 
 namespace facebook::velox::exec {
 
+// Represents arguments for window functions. Stores argument type,
+// the ConstantTypedExpr(if it is one) or the column index in the input row
+// (if the argument is a FieldAccessTypedExpr).
+struct WindowFunctionArg {
+  const TypePtr type;
+  const core::ConstantTypedExpr* constantValue;
```

Comment:
Any particular reason not to use ConstantVectorPtr here?

Reply:
I used ConstantTypedExpr as it looked more generic. But ConsantVectorPtr is easier to use and likely more suitable in a function API. Have changed that.